From 88266fed3fb4b0e681ad5a85a8889aa0c9ce4ad7 Mon Sep 17 00:00:00 2001
From: Fei Yang <fei.yang@intel.com>
Date: Wed, 25 Jun 2014 00:16:24 -0700
Subject: [PATCH 1/3] Check VBUS status

Need to test on real hardware before finalizing the code.

Change-Id: I56212210fe3185176144ae126b0ad5566fe9e935
Signed-off-by: Fei Yang <fei.yang@intel.com>
---
 drivers/power/bq24192_charger.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/power/bq24192_charger.c b/drivers/power/bq24192_charger.c
index df5771a..977b453 100644
--- a/drivers/power/bq24192_charger.c
+++ b/drivers/power/bq24192_charger.c
@@ -1038,7 +1038,18 @@ EXPORT_SYMBOL(bq24192_vbus_disable);
 int bq24192_vbus_status(void)
 {
 	struct bq24192_chip *chip = i2c_get_clientdata(bq24192_client);
-	return chip->a_bus_enable;
+	int val;
+
+	val = bq24192_read_reg(chip->client, BQ24192_SYSTEM_STAT_REG);
+	if (val < 0) {
+		dev_warn(&chip->client->dev, "System Status reg read fail\n");
+		return 0;
+	}
+	val &= SYSTEM_STAT_VBUS_BITS;
+	if (val == SYSTEM_STAT_VBUS_HOST || val == SYSTEM_STAT_VBUS_ADP)
+		return 1;
+	else
+		return 0;
 }
 EXPORT_SYMBOL(bq24192_vbus_status);
 
-- 
2.0.0

